node-agent/node_manager/src/lib.rs

```diff
@@ -1,3 +1,165 @@
-fn main() {
-    println!("Hello, world!");
+use std::{thread::sleep, time::Duration};
+
+use sysinfo::{CpuExt, DiskExt, System, SystemExt};
+
+const KB_TO_MB: u64 = 1000;
+const BIT_TO_GB: u64 = 1000000000;
+
+// used in all functions to access system
+fn system() -> System {
+    let mut sys = System::new_all();
+    sys.refresh_all();
+
+    sys
+}
```

Comment:
We're recreating a new `System` instance for every `system()` call, while the only thing we should be doing is refreshing it.

Please consider creating a static instance of `System` with e.g. the [lazy_static](https://crates.io/crates/lazy_static) crate instead and only refresh it when needed.

Reply:
Another option would be to have a `NodeSystem` structure:

```Rust
pub struct NodeSystem {
    inner: System,
}
```

define methods on it, and have your node manager instance carry one `NodeSystem` instance. It's probably a more elegant solution than a static `System` instance.

Reply:
You are right, I think I will change that with your second option.

Thank you.